src/azul/service/elasticsearch_service.py

```diff
@@ -328,26 +329,45 @@ def _generate_paging_dict(self, es_response, pagination):
         # ...else use search_after/search_before pagination
         es_hits = es_response['hits']['hits']
         count = len(es_hits)
+        self_url = pagination['self_url'] + f'?size={pagination["size"]}'
```

Comment:
Too pedestrian. Yes, size should be a string but you can't bet your life on it. It would be safer and more elegant to refactor this so there is exactly one place where ALL of the parameters are encoded and appended to the URL. Hint: Make `next_` and `previous` a dict compose the actual links as late as possible.